src/test/java/SpeechRankGUITests/src/main/java/driver/manager/BrowserFactory.java

```diff
@@ -0,0 +1,86 @@
+package driver.manager;
+
+import configuration.LocalWebDriverProperties;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.firefox.FirefoxOptions;
+import org.openqa.selenium.ie.InternetExplorerDriver;
+import org.openqa.selenium.ie.InternetExplorerOptions;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class BrowserFactory {
+
+    private static final String MESSAGE_UNKNOWN_BROWSER = "Unknown browser type! Please check your configuration";
+
+    private BrowserType browserType;
+
+    private boolean isRemoteRun;
+
+    public BrowserFactory(BrowserType browserType, boolean isRemoteRun) {
+        this.browserType = browserType;
+        this.isRemoteRun = isRemoteRun;
+    }
+
+    public WebDriver getBrowser() {
```

Comment:
why not to pass `BrowserType` and `isRemoteRun` args here, ger rid of the constructor and make this method static?